junit-jupiter-params/src/main/java/org/junit/jupiter/params/provider/MethodArgumentsProvider.java

```diff
@@ -104,10 +134,20 @@ private Method getFactoryMethodBySimpleName(Class<?> testClass, Method testMetho
 			// Otherwise, report that we didn't find anything.
 			return format("Could not find factory method [%s] in class [%s]", factoryMethodName, testClass.getName());
 		});
-		Preconditions.condition(factoryMethods.size() == 1,
-			() -> format("%d factory methods named [%s] were found in class [%s]: %s", factoryMethods.size(),
-				factoryMethodName, testClass.getName(), factoryMethods));
-		return factoryMethods.get(0);
+		return factoryMethods;
+	}
+
+	private static List<Method> filterFactoryMethodsWithMatchingParameters(List<Method> factoryMethods,
+			String factoryMethodName) {
+		if (!factoryMethodName.endsWith(")")) {
+			// If parameters are not specified, no choice is made
+			return factoryMethods;
+		}
+		String[] methodParts = ReflectionUtils.parseSimpleMethodName(factoryMethodName);
```

Comment:
I think we should avoid parsing the method name again here (in addition to in `findFactoryMethodCandidates`) and instead only do so once and pass it as an argument to both methods.